pkg/core/dao/dao.go

```diff
@@ -266,26 +268,67 @@ func (dao *Simple) AppendNEP5Transfer(acc util.Uint160, index uint32, tr *state.
 
 // -- start notification event.
 
-// GetAppExecResult gets application execution result from the
+// GetAppExecResults gets application execution results with the specified trigger from the
 // given store.
-func (dao *Simple) GetAppExecResult(hash util.Uint256) (*state.AppExecResult, error) {
-	aer := &state.AppExecResult{}
+func (dao *Simple) GetAppExecResults(hash util.Uint256, trig trigger.Type) ([]state.AppExecResult, error) {
 	key := storage.AppendPrefix(storage.STNotification, hash.BytesBE())
-	err := dao.GetAndDecode(aer, key)
+	aers, err := dao.Store.Get(key)
 	if err != nil {
 		return nil, err
 	}
-	return aer, nil
+	r := io.NewBinReaderFromBuf(aers[1:])
+	count := int(aers[0])
+	result := make([]state.AppExecResult, 0, count)
+	for i := 0; i < count; i++ {
```

Comment:
`for !r.IsEmpty() {`

```
func (r *BinReader) IsEmpty() bool {
        _, err := r.r.ReadByte()
        if err == io.EOF {
               return true
        } else if err == nil {
                r.r.UnreadByte()
        }
        return false
```

Reply:
We can't do this, because `r.r` is just an `io.Reader`. But the idea is implemented.